src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -511,6 +515,96 @@ private static ParsingStatus HexNumberToBigInteger(ref BigNumberBuffer number, o
             }
         }
 
+        private static ParsingStatus BinNumberToBigInteger(ref BigNumberBuffer number, out BigInteger result)
+        {
+            if (number.digits is null || number.digits.Length < 2)
+            {
+                result = default;
+                return ParsingStatus.Failed;
+            }
+
+            const int DigitsPerBlock = 32;
+
+            int totalDigitCount = number.digits.Length - 1;   // Ignore trailing '\0'
+
+            int blockCount = Math.DivRem(totalDigitCount, DigitsPerBlock, out int remainingDigitsInBlock);
+            if (remainingDigitsInBlock == 0)
+            {
+                remainingDigitsInBlock = DigitsPerBlock;
+            }
+            else
+            {
+                blockCount++;
+            }
+
+            Debug.Assert(number.digits[0] is '0' or '1');
+            bool isNegative = number.digits[0] == '1';
+
+            uint[]? arrayFromPool = null;
+            Span<uint> bufferSpan = blockCount <= BigIntegerCalculator.StackAllocThreshold ?
+                stackalloc uint[blockCount] : (arrayFromPool = ArrayPool<uint>.Shared.Rent(blockCount)).AsSpan(0, blockCount);
```

Comment:
nit: Could you separate this like it is in `HexNumberToBigInteger`:
```suggestion
            Span<uint> bufferSpan = ((uint)blockCount <= BigIntegerCalculator.StackAllocThreshold
                ? stackalloc uint[BigIntegerCalculator.StackAllocThreshold]
                : arrayFromPool = ArrayPool<uint>.Shared.Rent(blockCount)).Slice(0, blockCount);
```

It both helps with readability (clearly seeing the true/false path) and helps with perf because it avoids a dynamically sized stackalloc

Reply:
Changed, also changed similar code in `FormatBigIntegerToBin()`